projects/spectator/src/lib/spectator-routing/initial-module.ts

```diff
@@ -13,23 +16,41 @@ import { RouterLinkDirectiveStub } from './router-link-stub';
 export function initialRoutingModule<S>(options: Required<SpectatorRoutingOptions<S>>): ModuleMetadata {
   const moduleMetadata = initialSpectatorModule(options);
 
-  if (options.mockRouterLinks) {
+  if (options.mockRouterLinks && options.stubsEnabled) {
     moduleMetadata.declarations.push(RouterLinkDirectiveStub);
   }
 
-  moduleMetadata.providers.push(options.mockProvider(Router));
+  if (options.stubsEnabled) {
+    moduleMetadata.providers.push({
+      provide: Router,
+      useFactory: () => {
+        const provider = options.mockProvider(Router);
+        const router = provider.useFactory() as SpyObject<Router>;
 
-  moduleMetadata.providers.push([
-    {
-      provide: ActivatedRoute,
-      useFactory: () =>
-        new ActivatedRouteStub({
+        // this prevents the events property to be undefined
+        router.castToWritable().events = EMPTY;
```

Comment:
What if someone needs to test the router events? They might register a subscription in the component and want to test it.

Reply:
Yeah, for now I assume that you should provide your own mock provider if you want to configure mocks more advanced.

Reply:
Can't we just do the following?
```ts

options.mockProvider(Router, {
  events = new Subject();
  emitRouterEvent(e) {
    this.events.next(e)
  }
});
```

Reply:
all right, will do

Reply:
done, can you check again?